examples/matchmaker.rs

```diff
@@ -0,0 +1,110 @@
+use nakama_rs::{api_client::ApiClient, config};
+use nakama_rs::matchmaker::Matchmaker;
+use std::thread::sleep;
+use std::time::Duration;
+
+fn tick_clients(clients: &mut Vec<ApiClient>) {
+    while clients.iter().any(|client| client.in_progress()) {
+       clients.iter_mut().for_each(|client| {
+           client.tick();
+       })
+    }
+}
+
+fn wait_tokens(clients: &mut Vec<ApiClient>) {
+    while clients.iter().any(|client| client.matchmaker_token.is_none()) {
+        clients.iter_mut().for_each(|client| {
+            client.tick();
+        })
+    }
+}
+
+fn match_rank_range() {
+    let mut clients: Vec<ApiClient> = (1..5).map(|_| {
+        ApiClient::new("defaultkey", "127.0.0.1", config::DEFAULT_PORT, "http")
+    }).collect();
+
+    clients.iter_mut().enumerate().for_each(|(i, client)| {
+        client.register(&format!("email{}@provider.com", i), "password", &format!("Player{}", i));
+    });
+
+    tick_clients(&mut clients);
+
+    println!("Client 1 with rank 5 searching for a match with rank between 3 and 7");
+    let mut matchmaker = Matchmaker::new();
+    matchmaker
+        .add_numeric_property("rank", 5.0)
+        .add_query_item("rank").geq(3).required().build()
+        .add_query_item("rank").leq(7).required().build();
+
+    println!("Client 2 with rank 4 searching for a match with rank between 2 and 6");
+    let mut matchmaker2 = Matchmaker::new();
+    matchmaker2
+        .add_numeric_property("rank", 4.0)
+        .add_query_item("rank").geq(2).required().build()
+        .add_query_item("rank").leq(6).required().build();
+
+    println!("Client 3 with rank 10 searching for a match with rank between 8 and 12");
+    let mut matchmaker3 = Matchmaker::new();
+    matchmaker3
+        .add_numeric_property("rank", 10.0)
+        .add_query_item("rank").geq(8).required().build()
+        .add_query_item("rank").leq(12).required().build();
+
+    println!("Client 4 with rank 9 searching for a match with rank between 7 and 11");
+    let mut matchmaker4 = Matchmaker::new();
+    matchmaker4
+        .add_numeric_property("rank", 9.0)
+        .add_query_item("rank").geq(7).required().build()
```

Comment:
I am pretty sure that this should look like `add_query_item(QueryItemBuilder::new().whatever().build()).add_query_item(..)`